### PR TITLE
Docs update new loss functions and examples

### PR DESCRIPTION
**Pull request by jlema**
- Added new loss functions to and examples to docs
- Updated docstrings to reflect binary cross entropy loss naming
- Updated auto-generated docs from docstrings
- Fixed sphinx doc build warnings

### CODE REVIEW

**Review by nushib — APPROVED** [review]

**Review by Nking92 — APPROVED** [review]

**Review by ilmarinen — APPROVED** [review]
@jlema +1

**Review by KathleenW1 — CHANGES_REQUESTED** [review]
Lowercase "cross entropy" in documentation's non-code running text  (Ideally, "cross-entropy" should be hyphenated throughout all documentation; does introducing the hyphenation now create inconsistency with earlier documentation?)

**Review by KathleenW1 — APPROVED** [review]

**Review by ilmarinen — APPROVED** [review]
@jlema +1 looks good.
